File: javascript/pro_10.js
function solution(stones, k) {
    let answer = 0;
    if(stones.length<=k)    return Math.max.apply(null,stones);
    let start = 0;
    let end = 200000000;
    while(start<=end){
        let mid = parseInt((start+end)/2);
        let cnt=0;
        for(let i=0; i<stones.length; i++){
            if(stones[i]-mid<0){
                cnt++;
                if(cnt>=k)  break;
            }else{
                cnt=0;
            }
        }
        if(cnt>=k){
            end = mid-1;
        }else{
            start = mid+1;
            answer = mid;
        }
    }
    return answer;
}

File: cpp2/pro_10.cpp
//가로w 세로h
#include<iostream>

using namespace std;

int GCD(int a, int b){
    if(b==0)    return a;
    return GCD(b, a%b);
}

long long solution(int w,int h) {
    long long answer = 1;
    answer = (long long)w*h;
    //최대공약수로 나눈거 만큼이네 더하고 뺴기1 곱하기 최대공약수
    int gcd = GCD(w,h);
    int a = w/gcd;
    int b = h/gcd;
    answer-=(a+b-1)*gcd;
    return answer;
}